### PR TITLE
Timestamped model mixin

### PR DESCRIPTION
**Pull request by marcospri**
Using the existing definitions of created and & updated

Running alembic revision --autogenerate results in an empty migration.

There's more models that have `created` but not updated, does it make sense to have CreatedMixin?

### CODE REVIEW

File: lms/models/_mixins.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        sa.DateTime(),
+        server_default=sa.func.now(),
+        default=datetime.datetime.utcnow,
+        nullable=False,
+    )
+    updated = sa.Column(
+        sa.DateTime(),
+        default=datetime.datetime.utcnow,
+        server_default=sa.func.now(),
+        onupdate=sa.func.now(),
```
Comment (marcospri) [inline lms/models/_mixins.py:17]: Only change over the existing one is onupdate using func.now() vs datetime

**Review by seanh — APPROVED** [review]

File: lms/models/_mixins.py (hunk 1 above)
Comment (jon-betts) [reply lms/models/_mixins.py:17]: There's some prior art for this in `h`, but I don't think we should take that too seriously: https://github.com/hypothesis/h/blob/master/h/db/mixins.py

One thing to note is the mixed use of Python / Postgres functions for specifying the date. From what I can see (https://docs.sqlalchemy.org/en/14/core/defaults.html#client-invoked-sql-expressions) you can use the server side version for the `default` and `onupdate`. Without knowing the exact outcome, it seems better to have an identical mechanism for setting them all if we can.

This insulates us from any drift between Postgres and pPthon

File: lms/models/_mixins.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import sqlalchemy as sa
+
+
+class TimestampedModelMixin:
```
Comment (jon-betts) [inline lms/models/_mixins.py:4]: Super nit pick, but this isn't necessarily a model / the fact it's a model comes from elsewhere. So I think this is `TimestampedMixin` or `WithTimestamps` or something.

Comment (marcospri) [reply lms/models/_mixins.py:4]: :+1:  for something like `CreatedUpdatedMixin`

Comment (marcospri) [reply lms/models/_mixins.py]: @jon-betts We could do `CreatedMixin` and `UpdatedMixin` but I don't we need those independently as we might for Accessed, what do you think?

Comment (jon-betts) [reply lms/models/_mixins.py]: I think we should follow the usage. If created and updated always go together, then the class should too. I can't think of a good reason why we wouldn't want these both if we ever have either. If we do find that happens, it's easy to split up.

The `accessed` does seem like a different kettle of fish, as we have to code specifically to support it wherever it's used, so it makes sense to me for that not to be a part of it.

**Review by jon-betts — APPROVED** [review]
Yep, I think this looks good. Although I'll fully admit to be an "expert" on this in the way anyone is after 15 minutes poking about the SQLAlchemy docs.

**Pull request merged**